tests/unit/data_access/test_entity_queries.py

```diff
@@ -1,9 +1,39 @@
+import pytest
 from sqlalchemy.orm import Query
 from application.data_access.entity_queries import _apply_limit_and_pagination_filters
+from application.data_access.entity_queries import _apply_period_option_filter
 from application.db.models import EntityOrm
+from unittest.mock import MagicMock
 
 
 def test__apply_limit_and_pagination_filters_with_no_filters_applied():
     query = Query(EntityOrm)
     result = _apply_limit_and_pagination_filters(query, params={"dataset": "testing"})
     assert result._limit_clause is None
+
+
+@pytest.mark.parametrize(
+    "period, expected_count, end_date",
+    [
+        (["current"], 1, None),  # Test case for current entities
+        (["historical"], 1, "2023-01-01"),  # Past date for historical entities
+        (["all"], 1, None),  # Test case for no filter
+    ],
+)
+def test__apply_period_option_filter(mocker, period, expected_count, end_date):
+    mock_get_session = mocker.patch(
+        "application.routers.entity.get_session", return_value=MagicMock()
+    )
+    session = mock_get_session.return_value
+    entity = EntityOrm(end_date=end_date)
+    mock_query = MagicMock()
+    mock_query.filter.return_value = mock_query
+    mock_query.all.return_value = [entity]
+    session.query.return_value = mock_query
```

Comment:
There's a lot of mocking here so I'm wondering if it's actually testing the code.

For instance you're mocking the query.filter method which contains the code that you've changed - so wouldn't these tests pass before you even made your change (ie they aren't testing your change)? Correct me if I'm wrong.

If possible it might be better to have an integration test for _apply_period_option_filter that doesn't mock the query.filter or query.all method, so if somebody were to change what we pass in to the filter method the test flags it. 

I'm not super well versed in this repo so I'm not sure how feasible that is, but let me know if you're running into trouble